core/bin/oauth.rs

```diff
@@ -203,6 +204,84 @@ async fn connections_access_token(
     }
 }
 
+#[derive(Deserialize)]
+struct CredentialPayload {
+    provider: CredentialProvider,
+    content: serde_json::Map<String, serde_json::Value>,
+}
+
+async fn credentials_create(
+    State(state): State<Arc<OAuthState>>,
+    Json(payload): Json<CredentialPayload>,
+) -> (StatusCode, Json<APIResponse>) {
+    match Credential::create(state.store.clone(), payload.provider, payload.content).await {
+        Err(e) => error_response(
+            StatusCode::INTERNAL_SERVER_ERROR,
+            "internal_server_error",
+            "Failed to create credential",
+            Some(e),
+        ),
+        Ok(c) => match c.unseal_encrypted_content() {
+            Err(e) => error_response(
+                StatusCode::INTERNAL_SERVER_ERROR,
+                "internal_server_error",
+                "Failed to unseal encrypted content",
+                Some(e),
+            ),
+            Ok(content) => (
+                StatusCode::OK,
+                Json(APIResponse {
+                    error: None,
+                    response: Some(json!({
+                        "credential": {
+                            "credential_id": c.credential_id(),
+                            "created": c.created(),
+                            "provider": c.provider(),
+                            "content": content,
```

Comment:
do we need to return the clear content from this endpoint ?

Reply:
not necessarely, removing.